lib/tests/packages-from-directory/default.nix

```
@@ -0,0 +1,43 @@
+{ lib ? import ../../.
+, pkgs ? import ../../../. {}
+}:
+let
+  inherit (pkgs) stdenv callPackage writeTextFile;
+  src = ./test-data;
+  packages = lib.packagesFromDirectory {
+    inherit callPackage;
+    directory = src;
+  };
+  package = name:
+    writeTextFile {
+      inherit name;
+      text = name;
+    };
```

No need to involve derivations. The functions works for any Nix value, so you could just use the string directly.

```suggestion
    name
```

(and then some things may simplify)